sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +177,163 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument', 'SatelliteNumber']
+        chkattr = ["Time", "Instrument", "SatelliteNumber"]
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+
+    def _get_goes_sat_num(self, date):
+        """
+        Determines the best satellite number for a given date.
+
+        Parameters
+        ----------
+
+        date : `astropy.time.Time`
+            The date to determine which satellite is active.
+        """
+
+        # GOES-17 is operational but currently does not provide Level 2 data
+        # GOES-16 start date is based on the availability of regular level 1b data
+
+        suvi_operational = {
+            16: TimeRange("2018-06-01", parse_time("now")),
+        }
+
+        results = []
+        for sat_num in suvi_operational:
+            if date in suvi_operational[sat_num]:
+                # if true then the satellite with sat_num is available
+                results.append(sat_num)
+
+        if results:
+            # Return the newest satellite
+            return max(results)
+        else:
+            # if no satellites were found then raise an exception
+            raise ValueError(f"No operational SUVI instrument on {date.strftime(TIME_FORMAT)}")
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified time range.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+        level : str
+            The level of the data.
+        wavelength :
```

Comment:
This lacks a type?

Reply:
This is a namedtuple or a quantity. I'm not sure how how to document that since it expects the named tuple to be of a certain kind.